### PR TITLE
Upgrade to Node.js 18 and bump all NPM dev dependencies

### LINKED ISSUE #281: Upgrade to Node 18
## Details
This issue is for upgrading the Node version since 14 (current version) has reached the end of it’s [security support period](https://href.li/?https://endoflife.date/nodejs). Since we rely on WP and WC components we should probably follow along with the Gutenberg issue:

https://github.com/WordPress/gutenberg/issues/48588

Their current plan is to jump to Node 18.

This issue will also be helpful to collect any blockers which might be preventing us from upgrading the Node version.

### Describe the solution you'd like 
Update to Node 18

## Acceptance criteria

* [ ]  Node engines set to >= 18
* [ ] .nvmrc set to v18
* [ ] App compiles without errors with Node 18
* [ ] JS tests run correctly without errors with Node 18

### PR DESCRIPTION
**Pull request by eason9487**
### Changes proposed in this Pull Request:

Closes #281 

- Upgrade to use Node.js v18.
- Bump all dev dependencies to solve all vulnerabilities.
   - This reduces the number of npm vulnerabilities from 34 to 0. 🎉 
- Fix JS linting errors.

### Checks:

* [x] Does your code follow the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Detailed test instructions:

1. Set Node.js to v18.
2. Run `npm install` to see if it can finish without errors.
3. Run `npm audit` to check if it reports 0 vulnerabilities.
4. Check if `npm start`, `npm run dev`, `npm run build`, and `npm run lint:js` work well.
5. Go to the admin page of **WooCommerce > Settings > Integration > Google Analytics. Toggle the "Use Global Site Tag" checkbox to see if it can show/hide the "Set Domain Name" option.
6. Follow the same test steps in #343 to determine if the event tracking is sent with **Checkout** block.

### Changelog entry

- Dev - Upgrade to Node.js 18.
- Dev - Upgrade all NPM dev dependencies.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,15 +38,15 @@
     "prebuild": "rm -rf ./vendor"
   },
   "engines": {
-    "node": ">=12",
-    "npm": ">=6"
+    "node": ">=18",
+    "npm": ">=9"
```
Comment (puntope) [inline package.json:42]: Node v16 is still compatible and gives 0 issues as well.

**Comment by puntope** [comment]
Hi @eason9487 

Thanks for the work done, however I see Node v16 is still compatible so I wonder if we should allow that version in the engines.

**Comment by eason9487** [comment]
Hi @puntope, thanks for the review.

> however I see Node v16 is still compatible so I wonder if we should allow that version in the engines.

Node.js v16 has exceeded its end-of-life date, 11 Sep 2023, so there should be no need to keep v16.

File: package.json (hunk 1 above)
Comment (eason9487) [reply package.json:42]: Please refer to https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/345#issuecomment-1874827702.

**Review by puntope — APPROVED** [review]
✅  LGTM

Thanks @eason9487 for clarifying the npm v16 topic. This can be merged

**Pull request merged**